routing/router.go

```diff
@@ -532,23 +526,15 @@ func (r *ChannelRouter) Start() error {
 			// We create a dummy, empty payment session such that
 			// we won't make another payment attempt when the
 			// result for the in-flight attempt is received.
-			//
-			// PayAttemptTime doesn't need to be set, as there is
-			// only a single attempt.
 			paySession := r.cfg.SessionSource.NewPaymentSessionEmpty()
 
-			lPayment := &LightningPayment{
-				PaymentHash: payment.Info.PaymentHash,
-			}
-
-			// TODO(joostjager): For mpp, possibly relaunch multiple
-			// in-flight htlcs here.
-			var attempt *channeldb.HTLCAttemptInfo
-			if len(payment.Attempts) > 0 {
-				attempt = &payment.Attempts[0]
-			}
-
-			_, _, err := r.sendPayment(attempt, lPayment, paySession)
+			// Timeout doesn't need to be set, as there is
+			// only a single attempt. We also set a zero fee limit,
+			// as no more routes should be tried.
+			_, _, err := r.sendPayment(
+				payment.Info.Value, 0,
+				payment.Info.PaymentHash, 0, paySession,
```

Comment:
Yes, I think a `nil` value here for payment session would be nice

Reply:
I prefer not passing nil values

Reply:
I am still trying to understand the deeper reason for not wanting to pass nil values. It is a very common thing to do, much more common than passing an object that is internally deactivated by an `empty` bool.

Reply:
Not a deeper reason other than in the extreme case the user has to do nil checks everywhere a variable is being used, which is error prone, and not pretty. IMO we should minimize the use of nils, and in this case it is not really needed.

Reply:
As far as I can see, "everywhere" is only one location (the call to RequestRoute). But it is not an important issue. If nobody else complains, I am ok with leaving as is.